diff -ur ksh.orig/src/cmd/ksh93/Makefile ksh/src/cmd/ksh93/Makefile
--- ksh.orig/src/cmd/ksh93/Makefile	Tue Nov 21 13:40:10 2000
+++ ksh/src/cmd/ksh93/Makefile	Sun May 13 04:05:06 2001
@@ -33,19 +33,15 @@
 SHOPT_ESH == 1			/* emacs/gmacs edit mode */
 SHOPT_VSH == 1			/* vi edit mode */
 SHOPT_SPAWN == 			/* use spawnve for fork/exec */
-SHOPT_SUID_EXEC == 1		/* allow (safe) suid/sgid shell scripts */
+SHOPT_SUID_EXEC == 		/* allow (safe) suid/sgid shell scripts */
 if  CC.HOSTTYPE == "win32.*"
 	CCFLAGS = $(CC.OPTIMIZE) $(CC.DLL)
 	SHOPT_CRNL == 1		/* <cr><nl> is eqivalent to <nl> */
 	SHOPT_NOCASE == 1	/* do case insensitive file compare */
 end
+CCFLAGS = $(CC.OPTIMIZE) $(CC.DLL)
 
 /* use the following libraries only if they exist */
-LIBS_opt := +lm +ljobs +li
-if ! "$(-mam:N=static*)" && CC.HOSTTYPE != "sgi.mips*"
-	LIBS_opt += +lsocket +lnsl
-end
-LIBS_opt += +lintl -ldl
 $(LIBS_opt)	: .DONTCARE
 
 LDLIBRARIES = $(LIBS_opt)
@@ -61,12 +57,12 @@
 
 /* the linux.sparc linker still has some problems */
 
-$(SH) :: sh.1 pmain.c $(CC.HOSTTYPE:N=linux.sparc:?-ldll??) $(CC.HOSTTYPE:N=win32.*:?-lshell?+lshell?)
+$(SH) :: sh.1 pmain.c -ldll -lksh
 
 DATAFILES = limits.c msg.c strdata.c testops.c keywords.c options.c \
                  signals.c aliases.c builtins.c variables.c lexstates.c
 
-shell $(VERSION) :LIBRARY: shell.3 nval.3 alarm.c cd_pwd.c cflow.c deparse.c \
+ksh $(VERSION) :LIBRARY: shell.3 nval.3 alarm.c cd_pwd.c cflow.c deparse.c \
 	getopts.c hist.c misc.c print.c read.c sleep.c trap.c test.c \
 	typeset.c ulimit.c umask.c whence.c main.c \
 	arith.c args.c array.c completion.c defs.c edit.c expand.c \
@@ -89,7 +85,7 @@
 
 suid_exec ::	CCFLAGS=$(NODLL) suid_exec.c +last
 
-shcomp	::	CCFLAGS=$(NODLL) shcomp.c -lshell
+shcomp	::	CCFLAGS=$(NODLL) shcomp.c -lksh
 
 shcomp.o : _BLD_shell=
 
diff -ur ksh.orig/src/cmd/ksh93/OPTIONS ksh/src/cmd/ksh93/OPTIONS
--- ksh.orig/src/cmd/ksh93/OPTIONS	Wed Aug 19 21:20:57 1998
+++ ksh/src/cmd/ksh93/OPTIONS	Sun May 13 04:04:33 2001
@@ -16,6 +16,6 @@
 RAWONLY=0	# make viraw the only vi mode.  Sometimes set automatically
 SEVENBIT=0	# strip the eigth bit from characters
 SPAWN=0		# use libast spawnve() in place of fork/exec when possible
-SUID_EXEC=1	# execute /etc/suid_exec for suid, sgid script
+SUID_EXEC=0	# execute /etc/suid_exec for suid, sgid script
 TIMEOUT=0	# number of seconds for shell timeout, 0 disables feature
 VSH=1		# vi edit option
